beacon_node/http_api/src/broadcast_stream.rs

```diff
@@ -0,0 +1,39 @@
+use crate::Stream;
+use async_stream::try_stream;
+use std::pin::Pin;
+use tokio::sync::broadcast::error::RecvError;
+use tokio::sync::broadcast::Receiver;
+
+use std::task::{Context, Poll};
+
+/// A wrapper around [`Receiver`] that implements [`Stream`]. Achieved by using the [`async-stream`] crate.
+///
+/// [`Receiver`]: struct@tokio::sync::broadcast::Receiver
+/// [`Stream`]: trait@crate::Stream
+/// [`async-stream`]: https://docs.rs/async-stream
+pub struct BroadcastStream<T: Clone> {
+    inner: Pin<Box<dyn Stream<Item = Result<T, RecvError>> + Send + Sync>>,
+}
+
+impl<T: Clone + Unpin + 'static + Send + Sync> BroadcastStream<T> {
+    /// Create a new `BroadcastStream`.
+    pub fn new(mut rx: Receiver<T>) -> Self {
+        let stream = try_stream! {
+            loop {
+                let item = rx.recv().await?;
+                yield item;
+            }
+        };
+        Self {
+            inner: Box::pin(stream),
+        }
+    }
+}
```

Comment:
Notice that I have updated code after some PR review, perhaps you can update here as well? It might be a better implementation now. https://github.com/tokio-rs/tokio/pull/3384

Reply:
Thanks for pointing this out! and thanks a lot for adding the feature to tokio, it's much appreciated!